antarest/study/storage/rawstudy/raw_study_service.py

```diff
@@ -392,7 +395,21 @@ def unarchive(self, study: RawStudy) -> None:
             self.import_study(study, fh)
 
     def get_archive_path(self, study: RawStudy) -> Path:
-        return Path(self.config.storage.archive_dir / f"{study.id}.zip")
+        """
+        Get archive path of a study.
+
+        Args:
+            study: The study to get the archive path for.
+
+        Returns:
+            The full path of the archive file (zip or 7z).
+        """
+        archive_dir: Path = self.config.storage.archive_dir
+        for suffix in [".zip", ".7z"]:
+            path = archive_dir.joinpath(f"{study.id}{suffix}")
+            if path.is_file():
+                return path
+        return archive_dir.joinpath(f"{study.id}.7z")
```

Comment:
This line doesn't make any sense. Either we find the .7z file and return it at l. 411, or we don't and should throw some sort of error. Returning a path for a non-existent file is dangerous if the caller expects a path.

Reply:
There are two possible uses for this function: either to read an existing archive, or to create a new one. In the second case, the archive doesn't yet exist, and we supply an archive name in 7z format.

Reply:
I think this is indeed a source of confusion, and that it would be preferable to create two separate functions: one to search for a supposedly existing archive, the other to suggest a name for the creation of an archive.

Reply:
Avoir une méthode `find_study_path` pour trouver le chemin de l'archive en effectuant une détection automatique selon l'extension `.zip` ou `.7z`.
Avoir une autre méthode pour créer le chemin `create_archive_path`.